### DISCUSSION TITLE
OpenVPN port-forward with ProtonVPN

### DISCUSSION THREAD

**Discussion by tomatoandcake** [haugene/docker-transmission-openvpn#2596] [Q&A]
hi all,

is someone able to assist with creating a port-forward for ProtonVPN and OpenVPN?

I've looked through here: [https://protonvpn.com/support/port-forwarding-manual-setup/](url) and changed my username to have the addition of "+pmp" at the end.

I can successfully get transmission up and running through ProtonVPN using the "+pmp" in my username. 

However when I try and execute the `natpmpc` command (listed in the linked page) in my transmission container's console - nothing happens. 

From reading the article above - I am thinking I have to enter that command in the console and view the output to get the open port. 

does anyone else use ProtonVPN and have you been able to use it with port forwarding in this

**Comment by tomatoandcake** [comment]
I figured it out! :P

**Reply by pkishino** [reply]
feel free to share how you solved it :) someone might come looking for this, thanks

**Reply by IBNobody** [reply]
@tomatoandcake Any info on how you did it? Did you install something to the docker container to run `natpmpc`? Because the options I'm seeing are not scriptable, and this was the closest I came.

https://www.reddit.com/r/ProtonVPN/comments/10n0w0m/port_forwarding_on_debiadietpi_with_wireguard/

**Reply by IBNobody** [reply]
Sooooo... Here's my steps so far.

`sudo docker exec TransmissionVPN apt update` to update the package list.

`sudo docker exec TransmissionVPN apt get natpmpc` to get natpmpc.

Then you can play around with the commands listed here...

https://protonvpn.com/support/port-forwarding-manual-setup/

However, what I need to figure out is...

1. The magical grep command to get the port from the natpmpc command.
2. If I need to enable a chron job to keep the port alive.
3. How to update Transmission every time I kill the container so that it gets the correct port.

**Reply by atmaybury** [reply]
> Sooooo... Here's my steps so far.
> 
> `sudo docker exec TransmissionVPN apt update` to update the package list.
> 
> `sudo docker exec TransmissionVPN apt get natpmpc` to get natpmpc.
> 
> Then you can play around with the commands listed here...
> 
> https://protonvpn.com/support/port-forwarding-manual-setup/
> 
> However, what I need to figure out is...
> 
> 1. The magical grep command to get the port from the natpmpc command.
> 2. If I need to enable a chron job to keep the port alive.
> 3. How to update Transmission every time I kill the container so that it gets the correct port.
> 

Did you ever figure this out? I'm working on getting the same thing going at the moment.

**Comment by IBNobody** [comment]
> Did you ever figure this out? I'm working on getting the same thing going at the moment.

https://github.com/haugene/docker-transmission-openvpn/discussions/2633#discussioncomment-6210225

However, I ended up abandoning TransmissionVPN in favor of a 3-container solution. GlueTUN, qbittorrent, and qbittorrent-natmap. The downside to this approach is that it requires you to use SSH or a Task to start the containers (via docker-compose) since Synology Docker on DSM 6 does not allow chained/shared network interfaces.

- https://drfrankenstein.co.uk/step-3-setting-up-a-docker-bridge-network/
- https://drfrankenstein.co.uk/2022/09/26/qbittorrent-with-gluetun-vpn-in-docker-on-a-synology-nas/
- https://github.com/qdm12/gluetun/wiki/Custom-provider
- https://github.com/soxfor/qbittorrent-natmap

I loved TransmissionVPN, but it has been the part of my setup that I've had the most problems with in the 6 years I've used it. Orders of magnitude more than any of my other containers.

**Reply by atmaybury** [reply]
Thanks! This is a huge help.

**Comment by mark-monteiro** [comment]
In case anyone else stumbles across this in the future, you can follow the instructions here to enable port forwarding for ProtonVPN: https://haugene.github.io/docker-transmission-openvpn/provider-specific/#protonvpn
